twemproxy/tests/test_twemproxy.py

```diff
@@ -0,0 +1,156 @@
+    'client_err',
+    'client_connections',
+    'server_ejects',
+    'forward_error',
+    'fragments'
+])
+
+SERVER_STATS = set([
+    'in_queue',
+    'out_queue',
+    'in_queue_bytes',
+    'out_queue_bytes',
+    'server_connections',
+    'server_timedout',
+    'server_err',
+    'server_eof',
+    'requests',
+    'request_bytes',
+    'responses',
+    'response_bytes',
+])
+
+SC_TAGS = ['host:{}'.format(HOST), 'port:{}'.format(PORT), 'optional:tag1']
+
+
+@pytest.fixture
+def check():
+    check = Twemproxy('twemproxy', {}, {})
+    return check
+
+
+@pytest.fixture(scope="session")
+def spin_up_twemproxy():
+    """
+    Start a cluster with one master, one replica and one unhealthy replica and
+    stop it after the tests are done.
+    If there's any problem executing docker-compose, let the exception bubble
+    up.
+    """
+
+    env = os.environ
+
+    compose_file = os.path.join(HERE, 'compose', 'docker-compose.yaml')
+
+    env['DOCKER_COMPOSE_FILE'] = compose_file
+    env['DOCKER_ADDR'] = get_docker_hostname()
+    env['WAIT_FOR_IT_SCRIPT_PATH'] = _wait_for_it_script()
+    env['SETUP_SCRIPT_PATH'] = os.path.join(HERE, 'compose', 'setup.sh')
+
+    args = [
+        "docker-compose",
+        "-f", compose_file
+    ]
+
+    try:
+        subprocess.check_call(args + ["up", "-d"], env=env)
+        if not wait_for_cluster():
+            raise Exception("The cluster never came up")
+    except Exception:
+        cleanup_twemproxy(args, env)
+        raise
+
+    time.sleep(15)
+
+    yield
+    cleanup_twemproxy(args, env)
+
+
+def cleanup_twemproxy(args, env):
+    subprocess.check_call(args + ["down"], env=env)
+
+
+def wait_for_cluster():
+    for _ in xrange(0, 10):
+        res = None
+        try:
+            socket.getaddrinfo(HOST, PORT, 0, 0, socket.IPPROTO_TCP)
+            return True
+        except Exception as e:
+            log.debug("exception: {0} res: {1}".format(e, res))
+            time.sleep(5)
+
+    return False
+
+
+def _wait_for_it_script():
+    """
+    FIXME: relying on the filesystem layout is a bad idea, the testing helper
+    should expose its path through the api instead
+    """
+    dir = os.path.join(TESTS_HELPER_DIR, 'scripts', 'wait-for-it.sh')
+    return os.path.abspath(dir)
+
+
+@pytest.fixture
+def setup_request():
+    """
+    A request needs to be made in order for some of the data to be seeded
+    """
+    url = "http://{}:{}".format(HOST, PORT)
+    try:
+        requests.get(url)
+    except Exception:
+        pass
+
+
+def test_check(check, spin_up_twemproxy, setup_request, aggregator):
+    instance = {
+        'host': get_docker_hostname(),
+        'port': 6222,
+        'tags': ['optional:tag1']
+    }
+
+    check.check(instance)
+
+    for stat in GLOBAL_STATS:
+        aggregator.assert_metric("twemproxy.{}".format(stat), at_least=0)
+    for stat in POOL_STATS:
+        aggregator.assert_metric("twemproxy.{}".format(stat), at_least=1, count=1)
```

Comment:
It's a bit confusing that we're using `at_least` and `count` at the same time. Should this just be `count`?

Same thing for the `assert_metric` below.